front-end/src/app/shared/services/report.service.ts

```diff
@@ -3,18 +3,19 @@ import { Observable, map, tap } from 'rxjs';
 import { Store } from '@ngrx/store';
 import { setActiveReportAction } from 'app/store/active-report.actions';
 import { setCashOnHandAction } from 'app/store/cash-on-hand.actions';
-import { Report, CashOnHand } from '../interfaces/report.interface';
+import { CashOnHand } from '../interfaces/cash-on-hand.interface';
+import { Report } from '../models/report-types/report.model';
 import { TableListService } from '../interfaces/table-list-service.interface';
 import { ListRestResponse } from '../models/rest-api.model';
-import { F3xSummaryService } from './f3x-summary.service';
-import { F3xSummary } from '../models/f3x-summary.model';
+import { F3xReportService } from './f3x-report.service';
+import { F3xReport } from '../models/report-types/f3x-report.model';
 import { ApiService } from './api.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ReportService implements TableListService<Report> {
-  constructor(private apiService: ApiService, private f3xSummaryService: F3xSummaryService, private store: Store) {}
+  constructor(private apiService: ApiService, private f3xReportService: F3xReportService, private store: Store) {}
 
   public getTableData(pageNumber = 1, ordering = ''): Observable<ListRestResponse> {
```

Comment:
I'd like to restructure the services for the reports a bit and model it similarly to how we have organized the TransactionServices 

Basically, look at TransactionService on how to set up the generic getTableData(), get(), delete(), etc with an endpoint string definition in the transaction-type (We can put it in the report models). The F3xReportService and F24ReportService (this needs to be created) will inherit the ReportService and extend where necessary, like the F3xReportService will have the setStoreCashOnHand() method. Ping me if you have questions